packages/server/src/plugin/traceTreeBuilder.ts

```diff
@@ -280,6 +286,29 @@ function responsePathAsString(p?: ResponsePath): string {
   return res;
 }
 
+function responsePathFromArray(
+  pathAsArray: ReadonlyArray<{ key: string | number; typename?: string }>,
+): ResponsePath {
+  const idx = pathAsArray.length - 1;
+  return {
+    key: pathAsArray[idx].key,
+    prev: walkPath(pathAsArray, idx - 1),
+    typename: pathAsArray[idx].typename,
+  };
+}
```

Comment:
It was a little surprising not to find anything already doing this, is there a better way @glasser ?

Reply:
You can use graphql-js's `addPath` instead of constructing the object yourself, if that feels helpful.

You could also just not construct the intermediate array pathWithTypenames with map, and build up the response path bit by bit?  `responsePath = addPath(key, responsePath, typename)` or whatever?

Reply:
Much better! 1383a806522b17c6e94e7b846c1161a49f75c96a